nvm.sh

```diff
@@ -1231,6 +1231,25 @@ nvm_install_node_source() {
     make='gmake'
     MAKE_CXX="CXX=c++"
   fi
+
+  if [ -z "$NVM_CPU_THREADS" ] || [[ ! $NVM_CPU_THREADS =~ ^[0-9]+$ ]] ; then
+    if [ "_$NVM_OS" = "_linux" ]; then
+      NVM_CPU_THREADS="$(grep -c 'core id' /proc/cpuinfo)"
+    elif [ "_$NVM_OS" = "_freebsd" ] || [ "_$NVM_OS" = "_darwin" ]; then
+      NVM_CPU_THREADS="$(sysctl -n hw.ncpu)"
+    elif [ "_$NVM_OS" = "_sunos" ]; then
+      NVM_CPU_THREADS="$(psrinfo | wc -l)"
+    fi
+    if [ -z "$NVM_CPU_THREADS" ] || [[ ! $NVM_CPU_THREADS =~ ^[0-9]+$ ]] ; then
+      echo "Can not determine how many thread(s) we can use, set to only 1 now." 1>&2
+      echo "Please report an issue on GitHub to help us make it better and run it faster on your computer!" 1>&2
```

Comment:
If you want to encourage people to include `nvm debug` output in this error message (and then update `nvm debug` to output `NVM_OS` and `NVM_ARCH`), that would help the error reports be more useful :-)